File: 204-count-primes/count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {

        if (n<2) return 0;
        // marking all multiples from 2 to n as non-prime
        int cnt = 0;

        vector<bool>primes(n, true); // first mark all numbers as false
        primes[0] = primes[1] = false;
        
        for (int i=2; i*i < n; ++i){
            if (primes[i]) {
                // mark multples of i as non prime
                for (int multiple = i*i; multiple < n; multiple+=i){
                    primes[multiple] = false;
                }
            }
        }
        // count the no. of true values
        cnt = count(primes.begin(), primes.end(), true);
        return cnt;
    }
};

File: 204-count-primes/count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 2:
            return 0
        else:
            primes = [True] * n
            primes[0] = primes[1] = False


            i = 2

            while i*i < n:
                if primes[i]:
                    multiple = i * i

                    while multiple < n:
                        primes[multiple] = False
                        multiple += i
                i+=1

            count = 0
            for cond in primes:
                if cond==True:
                    count+=1
            
            return count
        